sql/expression/columndefault.go

```diff
@@ -0,0 +1,35 @@
+package expression
+
+import (
+	"fmt"
+	"github.com/liquidata-inc/go-mysql-server/sql"
+)
+
+// This function is to prevent import cycles. Since the sql package cannot reference the expression package, we have to
+// push the required function to a global that will be referenced from within the sql package. Otherwise, we'd need to
+// duplicate a lot of code or split up sql into smaller packages. If a better way is found, then feel free to replace this.
+func init() {
+	sql.ColumnDefaultManager.ExpressionTransformUp = func(inputExpr sql.Expression) (sql.Expression, error) {
+		return TransformUp(inputExpr, func(e sql.Expression) (sql.Expression, error) {
```

Comment:
Don't do this. If you need to resolve columns, functions, etc. the place to do so is in the analyzer, on an insert plan node that would use a default. If you want to just do validation (make sure every expression type is allowed), then use sql.Inspect.

Reply:
That works when you're on the standard SQL path, but if the idea is to have integrators store the default expression as a string, how would they be able to fetch the default outside of the SQL path, such as `dolt`'s import path? The current implementation is geared so that integrators can keep their non-SQL write paths while accessing SQL's concept of default values. For things like column values being referenced outside of the SQL path, I was imagining a callback of some sort that the integrator may provide for each column. This would just be a function call on the `*ColumnDefaultValue` type. If we changed the default value to a standard `sql.Expression`, then we could include the same on the `*Column` type itself.